# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/grub2/hotfix-riscv32.patch
# Copyright (C) 2022 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- grub-2.06/util/grub-mkimagexx.c.vanilla	2022-07-10 22:11:00.577549952 +0200
+++ grub-2.06/util/grub-mkimagexx.c	2022-07-10 22:11:39.600551965 +0200
@@ -1298,7 +1298,7 @@
 		       grub_uint32_t hi20, lo12;
 
 		       if (off != (grub_int32_t)off)
-			 grub_util_error ("target %lx not reachable from pc=%lx", (long)sym_addr, (long)((char *)target - (char *)e));
+			 grub_util_warn ("target %lx not reachable from pc=%lx", (long)sym_addr, (long)((char *)target - (char *)e));
 
 		       hi20 = (off + 0x800) & 0xfffff000;
 		       lo12 = (off - hi20) & 0xfff;
@@ -1337,7 +1337,7 @@
 		       grub_int32_t hi20;
 
 		       if (off != (grub_int32_t)off)
-			 grub_util_error ("target %lx not reachable from pc=%lx", (long)sym_addr, (long)((char *)target - (char *)e));
+			 grub_util_warn ("target %lx not reachable from pc=%lx", (long)sym_addr, (long)((char *)target - (char *)e));
 
 		       hi20 = (off + 0x800) & 0xfffff000;
 		       *t32 = grub_host_to_target32 ((grub_target_to_host32 (*t32) & 0xfff) | hi20);
